cmd/interceptors/main.go

```diff
@@ -82,13 +111,108 @@ func main() {
+			if _, err := tc.TriggersV1alpha1().ClusterInterceptors().Update(ctx, &clusterInterceptorList.Items[i], metav1.UpdateOptions{}); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
 }
 
 func handler(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
+
+// getCerts uses Knative pkg to generate certs for clusterinterceptor to run as https
+func getCerts(ctx context.Context, secretLister v1.SecretLister, kubeClient *kubeclientset.Clientset, logger *zap.SugaredLogger) (string, string, []byte, error) {
+	interceptorSvcName := os.Getenv("INTERCEPTOR_TLS_SVC_NAME")
+	interceptorSecretName := os.Getenv("INTERCEPTOR_TLS_SECRET_NAME")
+	namespace := os.Getenv("SYSTEM_NAMESPACE")
+
+	secret, err := secretLister.Secrets(namespace).Get(interceptorSecretName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// The secret should be created explicitly by a higher-level system
+			// that's responsible for install/updates.  We simply populate the
+			// secret information.
+			logger.Infof("secret %s is missing", interceptorSecretName)
+			return "", "", []byte{}, err
+		}
+		logger.Infof("error accessing certificate secret %q: %v", interceptorSecretName, err)
+		return "", "", []byte{}, err
+	}
+
+	var (
+		serverKey, serverCert, caCert []byte
+		createCerts                   bool
+	)
+	serverKey, haskey := secret.Data[certresources.ServerKey]
+	if !haskey {
+		logger.Infof("secret %q is missing key %q", secret.Name, certresources.ServerKey)
+		createCerts = true
+	}
+	serverCert, haskey = secret.Data[certresources.ServerCert]
+	if !haskey {
+		logger.Infof("secret %q is missing key %q", secret.Name, certresources.ServerCert)
+		createCerts = true
+	}
+	caCert, haskey = secret.Data[certresources.CACert]
+	if !haskey {
+		logger.Infof("secret %q is missing key %q", secret.Name, certresources.CACert)
+		createCerts = true
+	}
+
+	// TODO: Certification validation and rotation is pending
+
+	if createCerts {
+		serverKey, serverCert, caCert, err = certresources.CreateCerts(ctx, interceptorSvcName, namespace, time.Now().Add(decade))
+		if err != nil {
+			logger.Errorf("failed to create certs : %v", err)
+			return "", "", []byte{}, err
+		}
+
+		secret.Data = map[string][]byte{
+			certresources.ServerKey:  serverKey,
+			certresources.ServerCert: serverCert,
+			certresources.CACert:     caCert,
+		}
+		if _, err = kubeClient.CoreV1().Secrets(namespace).Update(ctx, secret, metav1.UpdateOptions{}); err != nil {
+			logger.Errorf("failed to update secret : %v", err)
+			return "", "", []byte{}, err
+		}
+	}
+
+	// write serverKey to file so that it can be passed while running https server.
+	if err = ioutil.WriteFile(keyFile, serverKey, 0600); err != nil {
+		logger.Errorf("failed to write serverKey file %v", err)
+		return "", "", []byte{}, err
+	}
+	// write serverCert to file so that it can be passed while running https server.
+	// Expect WriteFile permissions to be 0600 or less (gosec)
+	//nolint:gosec
```

Comment:
Hmm why do we have 0600?

Reply:
if given file permission is more than 600 the its giving security issue during build
```
cmd/interceptors/main.go:211:11: G306: Expect WriteFile permissions to be 0600 or less (gosec)
	if err = ioutil.WriteFile(certFile, serverCert, 0644); err != nil {
	         ^
```
For now instead of using 644 i m using 600 and removing `//nolint:gosec` 
Updated main.go 

Thanks for pointing it out :+1: I forgot to remove it earlier